connectors/protocol/connectors.py

```diff
@@ -738,24 +743,43 @@ async def sync_starts(self, job_type):
             msg = f"Unknown job type: {job_type}"
             raise ValueError(msg)
 
-        doc = {
-            "status": Status.CONNECTED.value,
-            "error": None,
-        } | last_sync_information
+        if self.index.feature_use_connectors_api:
+            await self.index.api.connector_update_last_sync_info(
+                connector_id=self.id, last_sync_info=last_sync_information
+            )
+            await self.index.api.connector_update_status(
+                connector_id=self.id, status=Status.CONNECTED.value
+            )
+            await self.index.api.connector_update_error(
+                connector_id=self.id, error=None
+            )
+        else:
+            doc = {
+                "status": Status.CONNECTED.value,
+                "error": None,
+            } | last_sync_information
 
-        await self.index.update(
-            doc_id=self.id,
-            doc=doc,
-            if_seq_no=self._seq_no,
-            if_primary_term=self._primary_term,
-        )
+            await self.index.update(
+                doc_id=self.id,
+                doc=doc,
+                if_seq_no=self._seq_no,
+                if_primary_term=self._primary_term,
+            )
 
     async def error(self, error):
-        doc = {
-            "status": Status.ERROR.value,
-            "error": str(error),
-        }
-        await self.index.update(doc_id=self.id, doc=doc)
+        if self.index.feature_use_connectors_api:
+            await self.index.api.connector_update_error(
+                connector_id=self.id, error=error
+            )
+            await self.index.api.connector_update_status(
+                connector_id=self.id, status=Status.ERROR.value
+            )
```

Comment:
Same here - this is an atomic action "mark as error" that updates status and writes error, should it be single action?

Reply:
Agreed that we could unify this in a following way:
- we just call `_error` endpoint, the logic in ES could set the status depending if `error` is null or not

Reply:
That would be perfect IMO!